autogluon_zeroshot/simulation/simulation_context.py

```diff
@@ -265,6 +273,14 @@ def subset_datasets(self, datasets):
                 raise ValueError(f'Missing expected dataset {d} in ZeroshotSimulatorContext!')
         self._update_unique_datasets(unique_datasets_subset)
 
+        # Remove datasets from internal dataframes
+        self.df_raw = self.df_raw[self.df_raw.tid.isin(datasets)]
+        self.df_results_by_dataset_vs_automl = self.df_results_by_dataset_vs_automl[self.df_results_by_dataset_vs_automl["tid"].isin(datasets)]
+        self.df_results_by_dataset_automl['tid'] = self.df_results_by_dataset_automl.apply(lambda x: int(x["dataset"].split("_")[0]), axis=1)
+        self.df_results_by_dataset_automl = self.df_results_by_dataset_automl[self.df_results_by_dataset_vs_automl["tid"].isin(datasets)]
```

Comment:
This line has a bug in it, it should be changed to:

```suggestion
        self.df_results_by_dataset_automl = self.df_results_by_dataset_automl[self.df_results_by_dataset_automl["tid"].isin(datasets)]
```

Reply:
Great catch, thanks.